### ISSUE TITLE
IDE0380 False Positive in New C# 16 Memory Model

### ISSUE THREAD

**Issue by wsugarman** [dotnet/roslyn#85732]
**Version Used**: VS Insiders [12210.170] / .NET SDK 11.0.100-rc.1.26425.128

**Steps to Reproduce**:

Program.cs
```csharp
using System;
using System.Runtime.InteropServices;

ReadOnlySpan<int> numbers = [1, 2, 3];
fixed (int* n = numbers)
{
    byte* b = (byte*)n;
    unsafe
    {
        Console.WriteLine(ReadInt(b));     // 1
        Console.WriteLine(ReadInt(b + 4)); // 2
        Console.WriteLine(ReadInt(b + 8)); // 3
        Console.WriteLine(ReadInt((byte*)0));
    }
}

static unsafe int ReadInt(byte* b)
{
    unsafe
    {
        ReadOnlySpan<byte> s = new(b, sizeof(int));
        return MemoryMarshal.Read<int>(s);
    }
}
```
.csproj
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
    <Features>$(Features);updated-memory-safety-rules</Features>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
    <LangVersion>preview</LangVersion>
    <Nullable>enable</Nullable>
    <OutputType>Exe</OutputType>
    <TargetFramework>net11.0</TargetFramework>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
    <WarningsAsErrors />
  </PropertyGroup>

</Project>
```

.editorconfig
```ini
[*.cs]
dotnet_analyzer_diagnostic.category-Style.severity = warning
dotnet_diagnostic.IDE0380.severity = error
```

**Diagnostic Id**: IDE0380: 'unsafe' modifier is unnecessary (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0380)

**Expected Behavior**: No warning with the new memory model as compilation is no longer the sole indicator for whether the `unsafe` keyword is appropriate

**Actual Behavior**: Warning

See Discussion here: https://github.com/dotnet/roslyn/discussions/85731
